File: Backend/ReStoreWebAPI/Entities/OrderAggregate/Order.cs
using System.ComponentModel.DataAnnotations;

namespace ReStoreWebAPI.Entities.OrderAggregate;

public class Order
{
    public int Id { get; set; }

    public string BuyerId { get; set; }

    [Required]
    public ShippingAddress ShippingAddress { get; set; }

    public DateTime Date { get; set; } = DateTime.Now;

    public List<OrderItem> Items { get; set; }

    public long Subtotal { get; set; }

    public long DeliveryFee { get; set; }

    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    public long GetTotal()
    {
        return Subtotal + DeliveryFee;
    }
}

File: Frontend/client/src/app/models/order.ts
export interface ShippingAddress {
    fullName: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    zip: string;
    country: string;
}

export interface OrderItem {
    productId: number;
    name: string;
    pictureUrl: string;
    price: number;
    quantity: number;
}
  
export interface Order {
    id: number;
    buyerId: string;
    shippingAddress: ShippingAddress;
    date: string;
    items: OrderItem[];
    subtotal: number;
    deliveryFee: number;
    status: string;
    totalPrice: number;
}